### PR TITLE
Add 'sort' method to Collections

### PR DESCRIPTION
**Pull request by rei2hu**
Sometimes people want to find the most recent or latest thing in a collection. This can be done with a single pass through the collection. However, sorting is easier to understand so here's a sorting method for collections that is based on mergesort.

### CODE REVIEW

File: src/util/Collection.js [hunk 1]
```diff
@@ -362,6 +362,61 @@ class Collection extends Map {
       return testVal !== value || (testVal === undefined && !collection.has(key));
     });
   }
+  
+  /**
+   * Sort a collection based on a function given
+   * @param {Function} fn Function used for comparing two objects
+   * @returns {Collection}
+   */
+   sort = (fn) {
```
Comment (Programmix) [inline src/util/Collection.js]: This line doesn't look right...
It should be `sort(fn) {`

Comment (rei2hu) [inline src/util/Collection.js]: You're right. I probably missed that when copying from prototype.sort = function(fn). Sorry about that.

**Comment by GusCaplan** [comment]
you can use `Map.prototype.entries` to your advantage to create a much faster and lighter sort using `Array.prototype.sort`. Tomorrow I can throw something together if you wish.

**Comment by GusCaplan** [comment]
i came up with this:
```js
sort(fn = (x, y) => x > y) {
  return new Collection(Array.from(this.entries()).sort((a, b) => fn(a[1], b[1], a[0], b[0])));
}
```
it lets you sort by keys and values for the map and only creates one new collection instead of like four for every time you can evenly divide the collection :)

**Comment by rei2hu** [comment]
I was trying to avoid converting to an array at first because I wasn't sure about the cost to do so vs making new objects in JS. After running a few tests today I guess it's a lot more expensive because your method is around 20x faster than this one for Collections of around size 25000.

**Pull request closed without merging**